src/ensembl/src/services/api-service.ts

```diff
@@ -67,12 +67,28 @@ class ApiService {
 
     try {
       const response = await fetch(url, fetchOptions);
+      if (!response.ok) {
+        throw await this.handleError(response, fetchOptions);
+      }
       return await this.handleResponse(response, fetchOptions);
     } catch (error) {
       throw error;
     }
   }
 
+  private async handleError(response: Response, options: FetchOptions) {
```

Comment:
Since `options` isn't used anywhere in this function can it be removed?

Reply:
removed